dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/DataOwnerApi.kt

```diff
@@ -0,0 +1,133 @@
+     */
+    @Operation(
+        summary = "Add a new data owner to a company.",
+        description = "A new data owner is added to the existing list for the specified company.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully added data owner."),
+        ],
+    )
+    @PostMapping(
+        produces = ["application/json"],
+        value = ["/{companyId}/data-owners/{userId}"],
+
+    )
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    fun postDataOwner(
+        @PathVariable("companyId") companyId: UUID,
+        @PathVariable("userId") userId: UUID,
+    ):
+        ResponseEntity<CompanyDataOwners>
+
+    /**
+     * A method to retrieve a  data owner information from companies in dataland
+     * @param companyId the ID of the company to which a new data owner is to be added
+     * @return userId of the data owner(s) of a specified company
+     */
+    @Operation(
+        summary = "Retrieve data owner(s) of a company.",
+        description = "Get a list of data owner(s) for the specified company.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved data owner."),
+            ApiResponse(responseCode = "404", description = "The specified company does not exist on Dataland."),
+        ],
+    )
+    @GetMapping(
+        produces = ["application/json"],
+        value = ["/{companyId}/data-owners"],
+
+    )
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    fun getDataOwners(
+        @PathVariable("companyId") companyId: UUID,
+    ):
+        ResponseEntity<List<String>>
+
+    /**
+     * A method to delete a data ownership relation in dataland
+     * @param companyId the ID of the company to which the data ownership should be removed from
+     * @param userId the ID of the user who is to be removed as company data owner
+     * @return information about the deleted company data ownership relation
+     */
+    @Operation(
+        summary = "Delete a data owner from a specified company.",
+        description = "An existing data owner is deleted from the existing list for the specified company.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully deleted data owner."),
+            ApiResponse(responseCode = "404", description = "The specified company doesn't exist on Dataland."),
+            ApiResponse(responseCode = "404", description = "Data owner doesn't exist for the specified company."),
+        ],
+    )
+    @DeleteMapping(
+        produces = ["application/json"],
+        value = ["/{companyId}/data-owners/{userId}"],
+
+    )
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    fun deleteDataOwner(
+        @PathVariable("companyId") companyId: UUID,
+        @PathVariable("userId") userId: UUID,
+    ):
+        ResponseEntity<CompanyDataOwners>
```

Comment:
![image](https://github.com/d-fine/Dataland/assets/104447806/f99c1206-01e1-44a3-ba67-af1188068852)
The 404 error is also thrown if the company does not exist. Maybe extend the description here. What do you think?

Reply:
Description should now be extended

Reply:
Leave this comment here for the delete due to the question whether to keep the error message, i.e. whether it is possible to print it without the JSON example output

Reply:
Error description was removed